### ISSUE TITLE
Issues with babel-preset-gatsby

### ISSUE THREAD

**Issue by Phil-Barber** [Js-Brecht/gatsby-plugin-ts-config#15]
Hey! 

Apologies if this is actually completely unrelated to this package but Installing this and making the recommended project structure changes are the only changes since the last successful run.

I'm getting this error after `gatsby clean && gatsby build`:

```
[gatsby-plugin-ts-config] An error occurred while reading your gatsby-node!                                                                                                                                 
[BABEL] /Users/phil/repos/phil-barber/.gatsby/gatsby-node.ts: `babel-preset-gatsby` has been loaded, which consumes config generated by the Gatsby CLI. Set `NODE_ENV=test` to bypass, or run `gatsby build`
 first. (While processing: "/Users/phil/repos/phil-barber/node_modules/babel-preset-gatsby/index.js")
```

My project structure is as mentioned in the README:
```
.
├── .gatsby
    ├── gatsby-config.ts
    ├── gatsby-node.ts
    ├── gatsby-browser.ts
    ├── gatsby-ssr..ts
    └── wrap-with-provider.tsx
└── gatsby-config.js
```

Here's my .babelrc
```
{
  "plugins": [
    "@babel/plugin-proposal-export-default-from",
    "babel-plugin-styled-components"
  ],
  "presets": [
    ["@babel/preset-typescript", { isTSX: true, allExtensions: true }],
    "babel-preset-gatsby"
  ]
}
```

tsconfig:
```
{
  "compilerOptions": {
    "outDir": "./dist/",
    "sourceMap": true,
    "noImplicitAny": true,
    "module": "commonjs",
    "target": "es2019",
    "jsx": "react",
    "lib": ["dom", "esnext"],
    "esModuleInterop": true,
    "skipLibCheck": true
  },
  "include": ["./src/**/*", ".gatsby/*"]
}
```

gatsby-config.js
```
const { generateConfig } = require('gatsby-plugin-ts-config');

module.exports = generateConfig({
  configDir: '.gatsby',
});

```

Let me know if there's any other info you need to help debug this. I'm trying to make a minimal reproduction from the gatsby-starter-typescript project for you but running into separate issues there so thought I'd post this first 🤦‍♂️

**Comment by Js-Brecht** [comment]
It’s because this plugin is spinning up a Babel process for transpiling, which causes it to read your `.baberc`.  `babel-preset-gatsby` is expecting to be run after bootstrap (during the Webpack phase), but this plugin is running before the bootstrap.

To fix this, you would need to use [onCreateBabelConfig](https://www.gatsbyjs.org/docs/node-apis/#onCreateBabelConfig) to load your `.babelrc` configurations manually.  You could likely store those settings next to `gatsby-node` (Name the file differently, though!), then read them in and loop over them, adding them to Gatsby using [setBabelPlugin](https://www.gatsbyjs.org/docs/actions/#setBabelPlugin) and [setBabelPreset](https://www.gatsbyjs.org/docs/actions/#setBabelPreset).  This should make the processing automatic, and you can maintain the settings file the same way you would `.babelrc`.

Out of curiosity, what issues were you having with `gatsby-preset-typescript`?

**Comment by Phil-Barber** [comment]
Thanks for the speedy reply!
That worked beautifully, cheers.

For posterity, the code added to `gatsby-node.ts` was:
```
import { CreateBabelConfigArgs } from 'gatsby';

exports.onCreateBabelConfig = ({ actions }: CreateBabelConfigArgs) => {
  actions.setBabelPreset({ name: 'babel-preset-gatsby', options: {} });
};
```
My babelrc is simple enough that I didn't bother with your more scalable suggestion but will bear it in mind going forward.

Much appreciated 👍 


Issue with `gatsby-starter-typescript` was:
```
The site's gatsby-config.js failed validation
ValidationError: "default" is not allowed
```
I'm sure it's just something trivial I'm missing (it usually is 😅 )

**Comment by Js-Brecht** [comment]
I’m glad that helped 🙂

Just as an FYI, Gatsby adds `babel-preset-gatsby` automatically.

---

There seems to be more `gatsby-starter-typescript` projects out there than I remember there being before.  Can you point me to the one you were using?  If there was no `export default` in the root `gatsby-config`, then that error may be caused by this plugin.  I’d like to see if there’s a good way to fix it

**Comment by Phil-Barber** [comment]
Here's the repo i was working on - but it's really probably not worth your time 😅 

https://github.com/Phil-Barber/gatsby-typescript-config

It's based off of:
https://github.com/goblindegook/gatsby-starter-typescript
Then I've installed this library and moved configs to `.gatsby/*`

**Comment by Js-Brecht** [comment]
Ah, okay, I figured out what was going on with it.  The Gatsby dependency version was `^2.0.7`.  I didn't look to see what the actual  version that was resolved by that spec was, but it was way too early.  Gatsby didn't support themes & default exports from `gatsby-config` in themes back then.

Additionally, the `configDir` from `generateConfig()` was spelled `.gastby`, which caused a few more issues.  That misspelling is actually very common; I can't count the number of times I've typed it like that, and it's one of those that's difficult to notice unless you're looking specifically for it.

After I fixed those two issues, it seemed like it started to work correctly.
